block/block_cache.go

```diff
@@ -0,0 +1,51 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/rollkit/rollkit/types"
+)
+
+type BlockCache struct {
+	blocks            map[uint64]*types.Block
+	hashes            map[string]bool
+	hardConfirmations map[string]bool
+	mtx               *sync.RWMutex
+}
```

Comment:
Great suggestion! However, this breaks the code: https://github.com/rollkit/rollkit/pull/1131

Do you have any idea why that is the case? Otherwise, we can skip this optimization for now 😃